### ISSUE TITLE
Release artifacts (models, dataset) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [X-GenGroup/PaCo-RL#1]
Hi @Jayce-Ping 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed that you have already released several excellent models (`PaCo-Reward-7B`, `PaCo-FLUX.1-dev`, `PaCo-FLUX.1-Kontext-dev`, `PaCo-QwenImage-Edit`) on the 🤗 Hub, which is fantastic! However, our automated system hasn't fully linked them to your paper page yet. It'd be great to fully integrate these checkpoints and the `PaCo-Reward Training Dataset` you mentioned in your paper into the 🤗 Hub's paper page for `https://huggingface.co/papers/2512.04784`, to improve their discoverability and visibility. We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make the dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by Jayce-Ping** [comment]
Hi Niels,


Thanks for your pull requests! I have merged those PRs.


The guidance in your email helps a lot! I need sometime to figure out how things work and I will add more usage code and model/dataset details later.&nbsp;


Thanks for helping! Cheers!


Bowen Ping




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "X-GenGroup/PaCo-RL"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年12月5日(星期五) 晚上6:27
***@***.***&gt;;
抄送:&nbsp;"Lazy ***@***.******@***.***&gt;;
主题:&nbsp;[X-GenGroup/PaCo-RL] Release artifacts (models, dataset) on Hugging Face (Issue #1)



NielsRogge created an issue (X-GenGroup/PaCo-RL#1)
 
Hi @Jayce-Ping 🤗
 
Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to hf.co/papers to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.
 
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
 the paper as yours which will show up on your public profile at HF, add Github and project page URLs.
 
I noticed that you have already released several excellent models (PaCo-Reward-7B, PaCo-FLUX.1-dev, PaCo-FLUX.1-Kontext-dev, PaCo-QwenImage-Edit) on the 🤗 Hub, which is fantastic! However, our automated system hasn't fully linked them to your paper page yet. It'd be great to fully integrate these checkpoints and the PaCo-Reward Training Dataset you mentioned in your paper into the 🤗 Hub's paper page for https://huggingface.co/papers/2512.04784, to improve their discoverability and visibility. We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.
 
Uploading models
 
See here for a guide: https://huggingface.co/docs/hub/models-uploading.
 
In this case, we could leverage the PyTorchModelHubMixin class which adds from_pretrained and push_to_hub to any custom nn.Module. Alternatively, one can leverages the hf_hub_download one-liner to download a checkpoint from the hub.
 
We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.
 
Uploading dataset
 
Would be awesome to make the dataset available on 🤗 , so that people can do:
 from datasets import load_dataset dataset = load_dataset("your-hf-org-or-username/your-dataset")
 
See here for a guide: https://huggingface.co/docs/datasets/loading.
 
Besides that, there's the dataset viewer which allows people to quickly explore the first few rows of the data in the browser.
 
Let me know if you're interested/need any help regarding this!
 
Cheers,
 
Niels
 ML Engineer @ HF 🤗
